src/plugins/core/pivot.ts

```diff
@@ -228,27 +246,73 @@ export class PivotCorePlugin extends CorePlugin<CoreState> implements CoreState
     formulaId = this.nextFormulaId.toString()
   ) {
     this.history.update("pivots", pivotId, { definition: deepCopy(pivot), formulaId });
-    this.compileCalculatedMeasures(pivot.measures);
+    this.compileCalculatedMeasures(pivotId, pivot.measures);
     this.history.update("formulaIds", formulaId, pivotId);
     this.history.update("nextFormulaId", this.nextFormulaId + 1);
   }
 
-  private compileCalculatedMeasures(measures: PivotCoreMeasure[]) {
+  private compileCalculatedMeasures(pivotId: UID, measures: PivotCoreMeasure[]) {
     for (const measure of measures) {
       if (measure.computedBy) {
-        const sheetId = measure.computedBy.sheetId;
         const compiledFormula = this.compileMeasureFormula(
           measure.computedBy.sheetId,
           measure.computedBy.formula
         );
         this.history.update(
           "compiledMeasureFormulas",
-          sheetId,
-          measure.computedBy.formula,
+          pivotId,
+          measure.id,
+          "formula",
           compiledFormula
         );
       }
     }
+    for (const measure of measures) {
+      if (measure.computedBy) {
+        const dependencies = this.computeMeasureFullDependencies(pivotId, measure);
+        this.history.update(
+          "compiledMeasureFormulas",
+          pivotId,
+          measure.id,
+          "fullDependencies",
+          dependencies
+        );
+      }
+    }
+  }
+
+  private computeMeasureFullDependencies(
+    pivotId: UID,
+    measure: PivotCoreMeasure,
+    exploredMeasures: Set<string> = new Set()
+  ): Range[] {
+    const rangeList: Range[] = [];
+    const definition = this.getPivotCoreDefinition(pivotId);
+    const formula = this.getMeasureCompiledFormula(pivotId, measure);
+    for (const token of formula.tokens) {
+      if (token.type !== "SYMBOL") {
+        continue;
+      }
+      const existingMeasure = definition.measures.find(
+        (measureCandidate) =>
+          getCanonicalSymbolName(measureCandidate.id) === token.value &&
+          measure.id !== measureCandidate.id
+      );
+
+      if (
+        !existingMeasure ||
+        exploredMeasures.has(existingMeasure.id) ||
+        !existingMeasure.computedBy
+      )
+        continue;
+
+      rangeList.push(
+        ...this.computeMeasureFullDependencies(pivotId, existingMeasure, exploredMeasures)
+      );
```

Comment:
this will create an infinite loop if we have cyclic dependencies in measures no ?